lsif/src/cache.test.ts

```diff
@@ -0,0 +1,145 @@
+import { GenericCache } from './cache'
+import * as sinon from 'sinon'
+
+describe('GenericCache', () => {
+    it('should evict items based by reverse recency', async () => {
+        const values = [
+            'foo', // foo*
+            'bar', // bar* foo
+            'baz', // baz* bar foo
+            'bonk', // bonk* baz bar foo
+            'quux', // quux* bonk baz bar foo
+            'bar', // bar quux bonk baz foo
+            'foo', // foo bar quux bonk baz
+            'honk', // honk* foo bar quux bonk
+            'foo', // foo honk bar quux bonk
+            'baz', // baz* foo honk bar quux
+        ]
+
+        // These are the cache values that need to be created, in-order
+        const expectedInstantiations = ['foo', 'bar', 'baz', 'bonk', 'quux', 'honk', 'baz']
+
+        let i = 0
+        const factory = sinon.stub()
+        for (const value of expectedInstantiations) {
```

Comment:
For when you need the index in an iteration
```suggestion
        for (const [i, value] of expectedInstantiations.entries()) {
```

Reply:
Done here and in another test.